### PR TITLE
Singing Machine

### PR DESCRIPTION
**Pull request by monochromatism**
## About The Pull Request

Hell week 2022! Adds Singing Machine, which is largely reworked from the original Lobotomy Corporation to be less awful in the context of LC13. Still pending balance, but testmerges should be harmless enough.
Also "adds" (technically) its armor, Harmony. Weapon code still pending.

## Why It's Good For The Game

More abnormalities! New armor (or armor that's now accessible consistently)! The start of a new framework for adding buff icons!

## Changelog
:cl:
add: Singing Machine
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/worktick(mob/living/carbon/human/user)
+	if(bonusRed) // If you have bonus red damage to apply...
+		user.apply_damage(bonusRed, RED_DAMAGE, null, user.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)
+		if(bonusRed < 6 && playStatus == 0)	// Should only happen when the machine isn't dealing damage.
+			for(var/mob/living/carbon/human/H in livinginrange(30, src))
+				if(faction_check_mob(H))
+					continue
+				H.adjustSanityLoss(rand(1,2))
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm:85]: This INCREASES sanity of people in a very big range. Is it the intended effect?

Comment (monochromatism) [reply code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm:85]: Yes, the intent is that when Singing Machine is "happy" its music is moderately helpful. The sanity healing can be toned down.

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
Atomize your changes.
This PR not only adds singing machine, but also modifies A LOT of other abnormalities.
Open new PRs for these changes.

**Comment by monochromatism** [comment]
What the hell? I just tried to rebase! Hang on, this should NOT have happened.

**Comment by monochromatism** [comment]
I'm just going to do the smart thing: start a brand new branch, copy every ACTUAL singing machine related change to it, and leave the rest of this to rot.

**Comment by monochromatism** [comment]
I really hope this is it.

**Comment by monochromatism** [comment]
No, it wasn't. Of course it wasn't. Why would it ever be.

**Pull request closed without merging**